yb-voyager/src/dbzm/valueConverter.go

```diff
@@ -0,0 +1,150 @@
+	} else {
+		return &NoOpValueConverter{}, nil
+	}
+}
+
+//============================================================================
+
+type NoOpValueConverter struct{}
+
+func (nvc *NoOpValueConverter) ConvertRow(tableName string, columnNames []string, row string) (string, error) {
+	return row, nil
+}
+
+func (nvc *NoOpValueConverter) ConvertEvent(ev *tgtdb.Event, table string) error {
+	return nil
+}
+
+//============================================================================
+
+type DebeziumValueConverter struct {
+	schemaRegistry      *SchemaRegistry
+	valueConverterSuite map[string]tgtdb.ConverterFn
+	converterFnCache    map[string][]tgtdb.ConverterFn
+	snapshotMode        bool
+}
+
+func NewDebeziumValueConverter(exportDir string, tdb tgtdb.TargetDB, snapshotMode bool) (*DebeziumValueConverter, error) {
+	schemaRegistry := NewSchemaRegistry(exportDir)
+	err := schemaRegistry.Init()
+	if err != nil {
+		return nil, fmt.Errorf("initializing schema registry: %w", err)
+	}
+	tdbValueConverterSuite := tdb.GetDebeziumValueConverterSuite(snapshotMode)
+
+	return &DebeziumValueConverter{
+		schemaRegistry:      schemaRegistry,
+		valueConverterSuite: tdbValueConverterSuite,
+		converterFnCache:    map[string][]tgtdb.ConverterFn{},
+		snapshotMode:        snapshotMode,
+	}, nil
+}
+
+func (conv *DebeziumValueConverter) ConvertRow(tableName string, columnNames []string, row string) (string, error) {
+	converterFns, err := conv.getConverterFns(tableName, columnNames)
+	if err != nil {
+		return "", fmt.Errorf("fetching converter functions: %w", err)
+	}
+	columnValues := strings.Split(row, "\t")
+	for i, columnValue := range columnValues {
+		if columnValue != "\\N" {
+			transformedValue, err := converterFns[i](columnValue, false)
+			if err != nil {
+				return "", fmt.Errorf("converting value: %w", err)
+			}
+			columnValues[i] = transformedValue
+		} else {
+			columnValues[i] = columnValue
+		}
+	}
+	return strings.Join(columnValues, "\t"), nil
+}
+
+func (conv *DebeziumValueConverter) getConverterFns(tableName string, columnNames []string) ([]tgtdb.ConverterFn, error) {
+	result := conv.converterFnCache[tableName]
+	if result == nil {
+		colTypes, err := conv.schemaRegistry.GetColumnTypes(tableName, columnNames)
+		if err != nil {
+			return nil, fmt.Errorf("fetching column schemas: %w", err)
+		}
+		result = make([]tgtdb.ConverterFn, len(columnNames))
+		for i := range columnNames {
+			colType := colTypes[i]
+			if conv.valueConverterSuite[colType] != nil {
+				result[i] = conv.valueConverterSuite[colType]
+			} else {
+				result[i] = func(v string, formatIfRequired bool) (string, error) { return v, nil }
+			}
+		}
+		conv.converterFnCache[tableName] = result
+	}
+	return result, nil
+}
+
+func (conv *DebeziumValueConverter) ConvertEvent(ev *tgtdb.Event, table string) error {
+	err := conv.convertMap(table, ev.Key)
+	if err != nil {
+		return fmt.Errorf("convert event fields: %w", err)
```

Comment:
On this line and line 123 swap `"fields"` and `"keys"` from the error messages.

Include event's sequenceID in the message. This comment applies to all such similar occurrences.